src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -5185,23 +5195,29 @@ private void PreVisitTypeArgAndProccesCallNode(CallNode node, TexlFunction func)
                     }
                     else
                     {
-                        _txb.ErrorContainer.Error(args[1], TexlStrings.ErrInvalidName, typeName.Ident.Name.Value);
-                        _txb.SetType(args[1], DType.Error);
+                        _txb.ErrorContainer.Error(typeArg, TexlStrings.ErrInvalidName, typeName.Ident.Name.Value);
+                        _txb.SetType(typeArg, DType.Error);
                         _txb.SetInfo(typeName, FirstNameInfo.Create(typeName, new NameLookupInfo(BindKind.NamedType, DType.Error, DPath.Root, 0)));
 
                         _txb.ErrorContainer.Error(node, TexlStrings.ErrInvalidArgumentExpectedType, typeName.Ident.Name.Value);
                         _txb.SetInfo(node, new CallInfo(func, node));
                         _txb.SetType(node, DType.Error);
                     }
                 }
-                else if (args[1] is TypeLiteralNode typeLiteral)
+                else if (typeArg is TypeLiteralNode typeLiteral)
                 {
                     VisitType(typeLiteral);
                 }
                 else
                 {
-                    _txb.ErrorContainer.Error(args[1], TexlStrings.ErrInvalidArgumentExpectedType, args[1]);
-                    _txb.SetType(args[1], DType.Error);
+                    _txb.ErrorContainer.Error(typeArg, TexlStrings.ErrInvalidArgumentExpectedType, typeArg);
+                    _txb.SetType(typeArg, DType.Error);
+                }
+
+                // Process other non-type arguments
+                for (var i = 1; i < args.Length - 1; i++)

```

Comment:
```                    if (overloadsWithTypeArgs.Any() && node.Args.Count > 1)
                    {
                        var nodeInp = node.Args.Children[0];
                        nodeInp.Accept(this);
                        Contracts.Assert(overloadsWithTypeArgs.Count() == 1);
                        var functionWithTypeArg = overloadsWithTypeArgs.First();
```

This bit seems odd (lines 4424-4425 earlier). Why do we visit the 1st arg first for these functions? Should we move that into the loop here as well?

Reply:
Actually the first argument visit is a requirement for the call to MatchOverloadWithUntypedOrJSONConversionFunctions. We probably shouldn't have that dependency, but I don't think I'll touch that code that is working right now